packages/utils/src/feature_toggle.rs

```diff
@@ -0,0 +1,414 @@
+                        "feature toggle: unknown feature '{}'",
+                        f
+                    )))
+                }
+                Some(s) => match s {
+                    FeatureStatus::Resumed => {}
+                    FeatureStatus::Stopped => {
+                        return Err(StdError::generic_err(format!(
+                            "feature toggle: feature '{}' is stopped",
+                            f
+                        )));
+                    }
+                },
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn handle_stop<S: Storage, A: Api, Q: Querier>(
+        deps: &mut Extern<S, A, Q>,
+        env: Env,
+        features: Vec<String>,
+    ) -> StdResult<HandleResponse> {
+        if Self::_get_pauser(&deps.storage, env.message.sender)?.is_none() {
+            return Err(StdError::unauthorized());
+        }
+
+        Self::_stop(&mut deps.storage, features)?;
+
+        Ok(HandleResponse {
+            messages: vec![],
+            log: vec![],
+            data: Some(to_binary(&HandleAnswer::Stop {
+                status: ResponseStatus::Success,
+            })?),
+        })
+    }
+
+    pub fn handle_resume<S: Storage, A: Api, Q: Querier>(
+        deps: &mut Extern<S, A, Q>,
+        env: Env,
+        features: Vec<String>,
+    ) -> StdResult<HandleResponse> {
+        if Self::_get_pauser(&deps.storage, env.message.sender)?.is_none() {
+            return Err(StdError::unauthorized());
+        }
+
+        Self::_resume(&mut deps.storage, features)?;
+
+        Ok(HandleResponse {
+            messages: vec![],
+            log: vec![],
+            data: Some(to_binary(&HandleAnswer::Resume {
+                status: ResponseStatus::Success,
+            })?),
+        })
+    }
+
+    pub fn handle_set_pauser<S: Storage, A: Api, Q: Querier>(
+        deps: &mut Extern<S, A, Q>,
+        _env: Env,
+        address: HumanAddr,
+    ) -> StdResult<HandleResponse> {
+        Self::_set_pauser(&mut deps.storage, address)?;
+
+        Ok(HandleResponse {
+            messages: vec![],
+            log: vec![],
+            data: Some(to_binary(&HandleAnswer::SetPauser {
+                status: ResponseStatus::Success,
+            })?),
+        })
+    }
+
+    pub fn handle_remove_pauser<S: Storage, A: Api, Q: Querier>(
+        deps: &mut Extern<S, A, Q>,
+        _env: Env,
+        address: HumanAddr,
+    ) -> StdResult<HandleResponse> {
+        Self::_remove_pauser(&mut deps.storage, address);
+
+        Ok(HandleResponse {
+            messages: vec![],
+            log: vec![],
+            data: Some(to_binary(&HandleAnswer::RemovePauser {
+                status: ResponseStatus::Success,
+            })?),
+        })
+    }
+
+    pub fn query_status<S: Storage, A: Api, Q: Querier>(
+        deps: &Extern<S, A, Q>,
+        features: Vec<String>,
+    ) -> QueryResult {
+        let mut status = vec![];
```

Comment:
This will prevent multiple allocations if many features are requested, and the vec would be grown multiple times
```rust
let mut status = vec::with_capacity(features.len());
```